Subject: [PATCH] testsuite: group T21035 compilation outputs

Each compilation produces several outputs. Ordinary multi-target rules
let parallel make run the same recipe more than once, causing duplicate
missing-library warnings or races renaming temporary object files.
Use grouped targets so each compilation runs once with MAKEFLAGS=-j30.

diff --git a/testsuite/tests/driver/T21035/Makefile b/testsuite/tests/driver/T21035/Makefile
index 3e426f7..bb7758e 100644
--- a/testsuite/tests/driver/T21035/Makefile
+++ b/testsuite/tests/driver/T21035/Makefile
@@ -8,10 +8,10 @@ BASE_VERSION = $('$GHC_PKG' field base id --simple-output)
 a.out: Main.o M.o
 	'$(TEST_HC)' Main.o M.o -package-env -
 
-Main.o Main.hi: M.hi hsdep/pkgdb/package.cache hsdep/HsDep.hi hsdep/libHShsdep-0.1-ghc8.10.7.so
+Main.o Main.hi &: M.hi hsdep/pkgdb/package.cache hsdep/HsDep.hi hsdep/libHShsdep-0.1-ghc8.10.7.so
 	'$(TEST_HC)' -c Main.hs hsdep/libHShsdep-0.1-ghc8.10.7.so -i. -package-env - -package-db hsdep/pkgdb
 
-M.o M.hi: M.hs hsdep-empty-lib/pkgdb/package.cache hsdep/HsDep.hi hsdep-empty-lib/libHShsdep-0.1-ghc8.10.7.so
+M.o M.hi &: M.hs hsdep-empty-lib/pkgdb/package.cache hsdep/HsDep.hi hsdep-empty-lib/libHShsdep-0.1-ghc8.10.7.so
 	'$(TEST_HC)' -c M.hs hsdep/HsDep.o -package-env - -package-db hsdep-empty-lib/pkgdb
 
 hsdep/pkgdb/package.cache: cat-hsdep-info.sh
@@ -33,7 +33,7 @@ hsdep-empty-lib/libHShsdep-0.1-ghc8.10.7.so:
 	'$(TEST_HC)' -shared -dynamic -o hsdep-empty-lib/libHShsdep-0.1-ghc8.10.7.so empty.c
 	rm empty.c
 
-hsdep/HsDep.dyn_hi hsdep/HsDep.dyn_o hsdep/HsDep.hi hsdep/HsDep.o: hsdep/HsDep.hs
+hsdep/HsDep.dyn_hi hsdep/HsDep.dyn_o hsdep/HsDep.hi hsdep/HsDep.o &: hsdep/HsDep.hs
 	'$(TEST_HC)' -c -dynamic-too -this-unit-id hsdep-0.1 hsdep/HsDep.hs -dynhisuf dyn_hi -dynosuf dyn_o
 
 T21035: a.out
